vpr/src/base/partition_regions.h

```diff
@@ -0,0 +1,37 @@
+#ifndef PARTITION_REGIONS_H
+#define PARTITION_REGIONS_H
+
+/**
+ * @file
+ * @brief This file defines the PartitionRegions class. The PartitionRegions class is used to store the union
+ * of regions that a particular partition can be placed in.
+ */
+
+class PartitionRegions {
+  public:
+    /**
+     * Returns the intersection of two PartitionRegions vectors that are passed to it.
+     */
+    PartitionRegions get_intersection(PartitionRegions region);
+
+    /**
+     * A test for intersection
+     * For example, test if the PartitionRegions of an atom intersect with PartitionRegions of a cluster.
+     */
+    bool intersects(PartitionRegions region);
+
+    /**
+     * Takes Atom Id and PartitonRegions vector and returns the intersection of the atom's PartitionRegions
+     * and the net PartitionRegions of the cluster (if any exists)
+     */
+    PartitionRegions get_intersection(const AtomBlockId& id, PartitionRegions part_regions);
```

Comment:
don't need the second argument - will only need atomblockid so you can get the part _regions of that, the other will come from the object you're calling for